src/dynamic_graph/dynamic-graph-py.cc

```diff
@@ -40,10 +46,24 @@ PyObject* plug(PyObject* /*self*/, PyObject* args) {
 
   pObjIn = PyCapsule_GetPointer(objIn, "dynamic_graph.Signal");
   SignalBase<int>* signalIn = (SignalBase<int>*)pObjIn;
-  if (signalIn == NULL) return NULL;
+  if (signalIn == NULL) {
+      struct module_state* st = GETSTATE(m);
+      std::ostringstream oss;
+      oss << "dynamic_graph.plug(a, b): Argument 'b' must be of type 'dynamic_graph.Signal', but got "
+          << PyCapsule_GetName(objIn);
+      PyErr_SetString(st->dgpyError, oss.str().c_str());
+      return NULL;
+  }
   pObjOut = PyCapsule_GetPointer(objOut, "dynamic_graph.Signal");
   SignalBase<int>* signalOut = (SignalBase<int>*)pObjOut;
-  if (signalOut == NULL) return NULL;
+  if (signalOut == NULL) {
+      struct module_state* st = GETSTATE(m);
+      std::ostringstream oss;
+      oss << "dynamic_graph.plug(a, b): Argument 'a' must be of type 'dynamic_graph.Signal', but got "
+          << PyCapsule_GetName(objOut);
+      PyErr_SetString(st->dgpyError, oss.str().c_str());
```

Comment:
Why not `PyErr_SetString(PyExc_TypeError, oss.str().c_str());` ?

Reply:
That would be better, indeed.